jets/c/muk.c

```diff
@@ -16,20 +16,27 @@
     c3_y *key_y;
     c3_w out_w;
 
-    c3_assert(u3r_met(5, seed) <= 1);
-    c3_assert(u3r_met(0, len)  <= 31);
-    c3_assert(u3r_met(3, key)  <= len);
+    c3_assert( u3r_met(5, seed) <= 1 );
+    c3_assert( u3r_met(0, len)  <= 31 );
+    c3_assert( u3r_met(3, key)  <= len );
 
     seed_w = u3r_word(0, seed);
     len_w  = u3r_word(0, len);
-    if (len_w > 0) { /* can't u3a_calloc 0 bytes */
-      key_y  = u3a_calloc(sizeof(c3_y), len);
-    } else {
+
+    //  can't u3a_calloc 0 bytes
+    //
+    if ( 0 == len_w ) {
       key_y = 0;
     }
-    u3r_bytes(0, len, key_y, key);
+    else {
+      key_y = u3a_calloc(sizeof(c3_y), len_w);
+    }
+
+    u3r_bytes(0, len_w, key_y, key);
 
-    MurmurHash3_x86_32(key_y, len, seed_w, &out_w);
+    //  XX we could skip the copy and use _mur_words() instead
+    //
+    MurmurHash3_x86_32(key_y, len_w, seed_w, &out_w);
```

Comment:
This issue is the same as the dichotomy and duplication of `u3r_mug_bytes` and `u3r_mug_words`. We could take advantage of the fact that a `u3a_atom` is an array of words and skip this copy. We would have to call the currently-static `_mur_words`, not u3r_mug_words, which is the 31-bit non-zero variant.

Note that this practically doesn't matter, as this jet is effectively unused. The only caller is `+mug`, and that is directly jetted with `u3r_mug`.

Reply:
Should we remove this jet, then?

Reply:
Oh, this is raw murmur3, which is +muk. I guess that's useful to have around.